shanoir-ng-datasets/src/main/java/org/shanoir/ng/dicom/web/service/DICOMWebService.java

```diff
@@ -100,33 +139,204 @@ public ResponseEntity findFrameOfStudyOfSerieOfInstance(String studyInstanceUID,
+		}
+		File[] dicomFiles = directoryWithDicomFiles.listFiles();
+		LOG.info("Start: STOW-RS sending " + dicomFiles.length + " dicom files to PACS from folder: " + directoryWithDicomFiles.getAbsolutePath());
+		MultipartEntityBuilder multipartEntityBuilder = MultipartEntityBuilder.create();
+		multipartEntityBuilder.setBoundary(BOUNDARY);
+		multipartEntityBuilder.setMimeSubtype(RELATED);
+		// create one multipart part for each file
+		for (File dicomFile : dicomFiles) {
+			FileInputStream fileIS = new FileInputStream(dicomFile);
+			// create content body
+			ContentBody contentBody = new InputStreamBody(
+					new ByteArrayInputStream(fileIS.readAllBytes()), ContentType.create(CONTENT_TYPE_DICOM));
+			// build MultipartPart
+			MultipartPartBuilder partBuilder = MultipartPartBuilder.create();
+			partBuilder.addHeader(CONTENT_TYPE, CONTENT_TYPE_DICOM);
+			partBuilder.setBody(contentBody);
+			MultipartPart multipartPart = partBuilder.build();
+			multipartEntityBuilder.addPart(multipartPart);
+		}
+		HttpEntity entity = multipartEntityBuilder.build();
+		sendMultipartRequest(entity);
+		LOG.info("Finished: STOW-RS sending " + dicomFiles.length + " dicom files to PACS from folder: " + directoryWithDicomFiles.getAbsolutePath());
+	}
+
+	@PreAuthorize("hasAnyRole('ADMIN', 'EXPERT', 'USER')")
+	public void sendDicomInputStreamToPacs(InputStream inputStream) throws Exception {
+		LOG.info("Start: STOW-RS sending dicom file input stream to PACS.");
+		// create content body
+		ContentBody contentBody = new InputStreamBody(
+				new ByteArrayInputStream(inputStream.readAllBytes()), ContentType.create(CONTENT_TYPE_DICOM));
+		// build MultipartPart
+		MultipartPartBuilder partBuilder = MultipartPartBuilder.create();
+		partBuilder.addHeader(CONTENT_TYPE, CONTENT_TYPE_DICOM);
+		partBuilder.setBody(contentBody);
+		MultipartPart multipartPart = partBuilder.build();
+		// build MultipartEntity
+		MultipartEntityBuilder multipartEntityBuilder = MultipartEntityBuilder.create();
+		multipartEntityBuilder.setBoundary(BOUNDARY);
+		multipartEntityBuilder.setMimeSubtype(RELATED);
+		multipartEntityBuilder.addPart(multipartPart);
+		HttpEntity entity = multipartEntityBuilder.build();
+		sendMultipartRequest(entity);
+		LOG.info("Finished: STOW-RS sending dicom file input stream to PACS.");
+	}
+
+	private void sendMultipartRequest(HttpEntity entity) throws IOException {
+		HttpPost httpPost = new HttpPost(dcm4cheeProtocol + dcm4cheeHost + ":" + dcm4cheePort + dicomWebRS);
+		httpPost.setHeader(HttpHeaders.CONTENT_TYPE, CONTENT_TYPE_MULTIPART+";type="+CONTENT_TYPE_DICOM+";boundary="+BOUNDARY);
+		httpPost.setEntity(entity);
+		try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
+			int code = response.getCode();
+			if (code != HttpStatus.OK.value()) {
+				LOG.error("DICOMWeb: sendMultipartRequest: response code not 200, but: " + code);				
+			}
+		} catch (IOException e) {
+			LOG.error(e.getMessage(), e);
+			throw e;
```

Comment:
I know you probably didn't touch this code, but the error/exception management is quite weird / un-logical along the class. (sometime we just log something, sometime we throw ShanoirException, sometime we rethrow the catched exception)

Reply:
Good remark: I will change it.